### PR TITLE
Bugfix/copy object bugs

### PR DESCRIPTION
**Pull request by shino**
This PR fixes a few bugs around PUT Copy / Multipart Upload Copy APIs.
- Fix bugs of the resource value in error response XML when source is
  not existent or not accessible, source bucket/key should be displayed
  there instead of target bucket/key.  The bug description is at
  https://github.com/basho/riak_cs/issues/1169, but this PR does not
  cover error response XML structure compatibility with S3.
- Fix Copy requests _without_ Content-Length request headers (https://github.com/basho/riak_cs/issues/939).
  Before this PR, such requests fails with 5xx errors. This PR allows
  requests without CL header in Copy API calls. Additionally, Copy API
  calls with CL more than zero is made to explicitly errors.
- Fix typo bug in upload ID seek, this makes error response of
  unknown upload ID correct (or more understandable.)

### CODE REVIEW

**Comment by kuenishi** [comment]
Rebasing or merging develop needed. 

``` diff
diff --cc riak_test/src/rtcs.erl
index 744446b,dccc445..0000000
--- a/riak_test/src/rtcs.erl
+++ b/riak_test/src/rtcs.erl
@@@ -1013,17 -1020,17 +1020,28 @@@ pbc({multibag, _} = Flavor, ObjectKind
      rtcs_bag:pbc(Flavor, ObjectKind, RiakNodes, Opts).

  make_authorization(Method, Resource, ContentType, Config, Date) ->
++<<<<<<< HEAD
 +    make_authorization(Method, Resource, ContentType, Config, Date, []).
 +
 +%% AmzHeaders are assumed already normalized and sorted
 +make_authorization(Method, Resource, ContentType, Config, Date, AmzHeaders) ->
 +    StsAmzHeaderPart = [[K, $:, V, $\n] || {K, V} <- AmzHeaders],
 +    StringToSign = [Method, $\n, [], $\n, ContentType, $\n, Date, $\n,
 +                    StsAmzHeaderPart, Resource],
 +    lager:debug("StringToSign~n~s~n", [StringToSign]),
++=======
+     make_authorization(s3, Method, Resource, ContentType, Config, Date).
+ 
+ make_authorization(Type, Method, Resource, ContentType, Config, Date) ->
+     StringToSign = [Method, $\n, [], $\n, ContentType, $\n, Date, $\n, Resource],
++>>>>>>> develop
      Signature =
          base64:encode_to_string(sha_mac(Config#aws_config.secret_access_key, StringToSign)),
-     lists:flatten(["AWS ", Config#aws_config.access_key_id, $:, Signature]).
+     Prefix = case Type of
+                  s3 -> "AWS";
+                  velvet -> "MOSS"
+              end,
+     lists:flatten([Prefix, " ", Config#aws_config.access_key_id, $:, Signature]).
```

**Comment by shino** [comment]
Rebased on current develop and force pushed.

File: src/riak_cs_wm_utils.erl [hunk 1]
```diff
@@ -1046,6 +1048,49 @@ check_timeskew(ReqTimestamp) when is_tuple(ReqTimestamp)->
...
+    case wrq:get_req_header("Content-Length", RD) of
+        undefined -> undefined;
+        CL ->
+            case (catch list_to_integer(CL)) of
+                Length when is_integer(Length) andalso 0 =< Length -> Length;
+                _Other -> {error, CL}
+            end
+    end.
+
+%% `valid_entity_length' helper.
```
Comment (kuenishi) [inline src/riak_cs_wm_utils.erl:1062]: It would be good to have `@doc` prefix to let edoc know.

**Comment by shino** [comment]
@borshop merge

**Pull request merged**